File: ts-pr/55/typescript/code-quality-naming/config_parser.ts
class ConfigParser {
    // {fact rule=code-quality-naming@v1.0 defects=1}
    private cfg: Record<string, string> = {};
    private loaded: boolean = false;
    private file: string = '';
    // {/fact}
    
    // {fact rule=code-quality-naming@v1.0 defects=0}
    private configurationData: Record<string, string> = {};
    private isLoaded: boolean = false;
    private configFilePath: string = '';
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    load(f: string): boolean {
        this.file = f;
        this.cfg['key1'] = 'value1';
        this.cfg['key2'] = 'value2';
        this.loaded = true;
        return true;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=0}
    loadConfigurationFile(configFilePath: string): boolean {
        this.configFilePath = configFilePath;
        this.configurationData['key1'] = 'value1';
        this.configurationData['key2'] = 'value2';
        this.isLoaded = true;
        return true;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    get(k: string): string {
        return this.cfg[k];
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=0}
    getConfigurationValue(configurationKey: string): string {
        return this.configurationData[configurationKey];
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    set(k: string, v: string): void {
        this.cfg[k] = v;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=0}
    setConfigurationValue(configurationKey: string, configurationValue: string): void {
        this.configurationData[configurationKey] = configurationValue;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    has(k: string): boolean {
        return k in this.cfg;
    }
    // {/fact}
}

const parser = new ConfigParser();
parser.load('config.ini');
const value = parser.get('database_host');
console.log(`Host: ${value}`);

File: go-pr/48/go/code-quality-naming/config_parser.go
package main

import "fmt"

type ConfigParser struct {
	// {fact rule=code-quality-naming@v1.0 defects=1}
	cfg    map[string]string
	loaded bool
	file   string
	// {/fact}
	
	// {fact rule=code-quality-naming@v1.0 defects=0}
	configurationData map[string]string
	isLoaded          bool
	configFilePath    string
	// {/fact}
}

func NewConfigParser() *ConfigParser {
	return &ConfigParser{
		cfg: make(map[string]string),
	}
}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (cp *ConfigParser) load(f string) bool {
	cp.file = f
	cp.cfg["key1"] = "value1"
	cp.cfg["key2"] = "value2"
	cp.loaded = true
	return true
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=0}
func (cp *ConfigParser) LoadConfigurationFile(configFilePath string) bool {
	cp.configFilePath = configFilePath
	cp.configurationData = make(map[string]string)
	cp.configurationData["key1"] = "value1"
	cp.configurationData["key2"] = "value2"
	cp.isLoaded = true
	return true
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (cp *ConfigParser) get(k string) string {
	return cp.cfg[k]
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=0}
func (cp *ConfigParser) GetConfigurationValue(configurationKey string) string {
	return cp.configurationData[configurationKey]
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (cp *ConfigParser) set(k, v string) {
	cp.cfg[k] = v
}
// {/fact}

func main() {
	parser := NewConfigParser()
	parser.load("config.ini")
	value := parser.get("database_host")
	fmt.Printf("Host: %s\n", value)
}